src/main/java/org/mskcc/cbio/oncokb/web/rest/SlackController.java

```diff
@@ -53,6 +56,19 @@ public SlackController(UserService userService, UserRepository userRepository, M
         this.userMapper = userMapper;
     }
 
+    private void updateUserWithRoleApiIfRequested(UserDTO userDTO) {
+        boolean apiAccessRequested = userDTO.getAdditionalInfo() != null && userDTO.getAdditionalInfo().getApiAccessRequest() != null && userDTO.getAdditionalInfo().getApiAccessRequest().isRequested();
+        if (userDTO.getLicenseType() != LicenseType.ACADEMIC || apiAccessRequested) {
+            Set<String> userDTOAuthorities = userDTO.getAuthorities();
+            userDTOAuthorities.add(AuthoritiesConstants.API);
+
+            if (userDTO.getLicenseType() == LicenseType.ACADEMIC) {
+                ApiAccessRequest userApiAccessRequest = userDTO.getAdditionalInfo().getApiAccessRequest();
+                userApiAccessRequest.setRequested(false);
```

Comment:
Minor comment, but probably don't need to create an extra variable here

`userDTO.getAdditionalInfo().getApiAccessRequest().setRequested(false)`